### PR TITLE
[FIX] web: kanban: display no content helper

### PR DESCRIPTION
**Pull request by aab-odoo**
The no content helper logic (determining when it is displayed) in kanban is quite complicated. Appearently, in a kanban view grouped on a field that isn't a many2one, with no groups (and thus no records), and where we don't have the "group_create" permission, we want the no content helper to be displayed. At least, this is how is worked in the legacy kanban view. This commit restores the previous behavior and encodes this situation in a test. In the future, it would probably be a good idea to sit down and challenge the corner cases of this feature.

### CODE REVIEW

File: addons/web/static/src/views/kanban/kanban_renderer.js [hunk 1]
```diff
@@ -226,14 +226,14 @@ export class KanbanRenderer extends Component {
             return true;
         }
         if (isGrouped) {
-            if (!this.state.columnQuickCreateIsFolded) {
+            if (this.canCreateGroup() && !this.state.columnQuickCreateIsFolded) {
```
Comment (Polymorphe57) [inline addons/web/static/src/views/kanban/kanban_renderer.js:229]: commit message:

Apparently
how is worked --> how it used to work

File: addons/web/static/src/views/kanban/kanban_renderer.xml [hunk 2]
```diff
@@ -143,24 +143,24 @@
                         exampleData="exampleData"
                         groupByFieldString="props.list.groupByField.string"
                     />
-                </t>
```
Comment (Polymorphe57) [inline addons/web/static/src/views/kanban/kanban_renderer.xml:146]: why the change in t position?

File: addons/web/static/src/views/kanban/kanban_renderer.js (hunk 1 above)
Comment (Polymorphe57) [inline addons/web/static/src/views/kanban/kanban_renderer.js:229]: a || instead?

**Comment by Polymorphe57** [comment]
robodoo r+

**Comment by aab-odoo** [comment]
robodoo r-

**Review by aab-odoo — COMMENTED** [review]
robodoo r+
